tests/phpunit/tests/widgets.php

```diff
@@ -686,6 +686,61 @@ public function test_wp_widget_get_settings() {
 		$this->assertArrayNotHasKey( 0, $never_used );
 	}
 
+	/**
+	 * @ticket 54677
+	 *
+	 * @covers WP_Widget::get_settings
+	 */
+	public function test_wp_widget_initializes_widget_with_alt_option() {
+		/*
+		 * Emulate a new the recent posts widget.
+		 *
+		 * The widget contains an alternative (legacy) option so both the
+		 * current and the alternative option need to be deleted.
+		 */
+		delete_option( 'widget_recent-posts' );
+		delete_option( 'widget_recent_entries' );
+
+		$this->assertSame( false, get_option( 'widget_recent-posts' ), 'The option widget_recent-posts was not deleted.' );
+		$this->assertSame( false, get_option( 'widget_recent_entries' ), 'The option widget_recent_entries was not deleted.' );
+
+		wp_widgets_init();
+		$this->assertSameSetsWithIndex( array( '_multiwidget' => 1 ), get_option( 'widget_recent-posts' ), 'Option failed to be initialized.' );
+		$this->assertFalse( get_option( 'widget_recent_entries' ), 'Alternative option is set.' );
+	}
+
+	/**
+	 * @ticket 54677
+	 *
+	 * @covers WP_Widget::get_settings
```

Comment:
```suggestion
	 * @covers ::get_settings
```

Reply:
Same as above. This suggestion is wrong.